src/driver.rs

```diff
@@ -410,6 +410,90 @@ impl DriverManager {
         Ok(Driver { c_driver })
     }
 
+    /// Get the Driver based on the file extension from filename
+    ///
+    /// Searches for the available extensions in the registered
+    /// drivers and returns the matches. The determined driver is
+    /// checked for writing capabilities as
+    /// [`Dataset::open`](Dataset::open) can already open datasets
+    /// with just path.
+    ///
+    /// See also: [`get_driver_by_name`](Self::get_driver_by_name)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let compatible_drivers =
+    ///     DriverManager::get_drivers_for_filename("test.gpkg", true)
+    ///         .iter()
+    ///         .map(|d| d.short_name())
+    ///         .collect::<Vec<String>>();
+    /// println!("{:?}", compatible_drivers);
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// ["GPKG"]
+    /// ```
+    pub fn guess_drivers_for_write(filename: &str, is_vector: bool) -> Vec<Driver> {
+        let ext = {
+            let filename = filename.to_ascii_lowercase();
+            let e = match filename.rsplit_once('.') {
+                Some(("", _)) => "", // hidden file no ext
+                Some((f, "zip")) => {
+                    // zip files could be zipped shp or gpkg
+                    if f.ends_with(".shp") {
+                        "shp.zip"
+                    } else if f.ends_with(".gpkg") {
+                        "gpkg.zip"
+                    } else {
+                        "zip"
+                    }
+                }
+                Some((_, e)) => e, // normal file with ext
+                None => "",
+            };
+            e.to_string()
+        };
+
+        let mut drivers: Vec<Driver> = Vec::new();
+        for i in 0..DriverManager::count() {
```

Comment:
Note for another PR: we should have an iterator over the available drivers.

Reply:
You mean to say we don't, but we should make one? Then yeah definitely. Can work on that.

If you mean we already have one, then I can edit it to use that.

Reply:
Don't think we do. In the meanwhile you could rewrite it with iterators, so it doesn't need to return a vector, but it's not super important.